ice/di.zep

```diff
@@ -271,30 +271,40 @@ class Di extends Arr
      *
      * @param string name The hook name
      * @param mixed callable A callable object
+     * @param mixed context The callback bind to the object
      * @param int priority The hook priority; 0 = high, 10 = low
+     * @return object Di
      */
-    public function hook(string name, var callback, int priority = 10)
+    public function hook(string name, var callback, var context = null, int priority = 10)
     {
-        if !isset this->hooks[name] {
-            let this->hooks[name] = [[]];
-        }
-
         if typeof callback == "callable" {
+            if context != null && is_object(context) {
+                let name = name . spl_object_id(context);
```

Comment:
`spl_object_id` function is added on php 7.2.0, maybe we could use `spl_object_hash`? 
```zep
let name .= '.' . spl_object_hash(context);
```

Reply:
good catch！